v2/concordium/types.proto

```diff
@@ -2101,20 +2140,27 @@ message PoolCurrentPaydayInfo {
 
 // Type for the response of GetPoolInfo.
 // Contains information about a given pool at the end of a given block.
+// From protocol version 7, pool removal has immediate effect, however, the
+// pool may still be present for the current (and possibly next) reward period.
+// In this case, the `current_payday_infor` field will be set, but the
+// `equity_capital`, `delegated_capital`, `delegated_capital_cap` and,
+// `pool_info` fields will all be absent. The `equity_pending_change` field
+// will also be absent, as stake changes are immediate.
 message PoolInfoResponse {
   // The 'BakerId' of the pool owner.
   BakerId baker = 1;
   // The account address of the pool owner.
   AccountAddress address = 2;
   // The equity capital provided by the pool owner.
-  Amount equity_capital = 3;
+  optional Amount equity_capital = 3;
   // The capital delegated to the pool by other accounts.
-  Amount delegated_capital = 4;
+  optional Amount delegated_capital = 4;
   // The maximum amount that may be delegated to the pool, accounting for leverage and stake limits.
-  Amount delegated_capital_cap = 5;
+  optional Amount delegated_capital_cap = 5;
   // The pool info associated with the pool: open status, metadata URL and commission rates.
-  BakerPoolInfo pool_info = 6;
-  // Any pending change to the equity carpital.
+  optional BakerPoolInfo pool_info = 6;
```

Comment:
**Suggestion (non-blocking):** Add when these are not present for each of the fields.

I know that this is already mentioned above, but since these comments are typically used for documentation in the SDKs (both the generated code, but we also tend to copy this everywhere) I think it would be good to have it here as well.
Also, how they behave prior to P7.